File: Algorithms/Search/KnightL on a Chessboard.py
import os

#
# Complete the 'knightlOnAChessboard' function below.
#
# The function is expected to return a 2D_INTEGER_ARRAY.
# The function accepts INTEGER n as parameter.
#

def knightlOnAChessboard(n):
    # Write your code here
    def bfs(a, b):
        queue = [(0, 0, 0)]
        visited = set()
        while queue:
            x, y, steps = queue.pop(0)
            if x == n - 1 and y == n - 1:
                return steps
            for dx, dy in [
                (a, b),
                (-a, b),
                (a, -b),
                (-a, -b),
                (b, a),
                (-b, a),
                (b, -a),
                (-b, -a),
            ]:
                nx, ny = x + dx, y + dy
                if 0 <= nx < n and 0 <= ny < n and (nx, ny) not in visited:
                    visited.add((nx, ny))
                    queue.append((nx, ny, steps + 1))
        return -1

    result = [[0] * (n - 1) for _ in range(n - 1)]
    for a in range(1, n):
        for b in range(1, n):
            result[a - 1][b - 1] = bfs(a, b)

    return result


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    result = knightlOnAChessboard(n)

    fptr.write('\n'.join([' '.join(map(str, x)) for x in result]))
    fptr.write('\n')
    fptr.close()

File: Algorithms/Search/KnightL on a Chessboard.php
<?php

/*
 * Complete the 'knightlOnAChessboard' function below.
 *
 * The function is expected to return a 2D_INTEGER_ARRAY.
 * The function accepts INTEGER n as parameter.
 */

function knightlOnAChessboard($n) {
    # Write your code here
    $result = [];

    for ($i = 1; $i < $n; $i++) {
        $row = [];
        for ($j = 1; $j < $n; $j++) {
            $moves = bfs($n, $i, $j);
            $row[] = $moves;
        }
        $result[] = $row;
    }

    return $result;
}

function bfs($n, $a, $b) {
    $queue = [[0, 0, 0]];
    $visited = array_fill(0, $n, array_fill(0, $n, False));

    while (!empty($queue)) {
        [$x, $y, $moves] = array_shift($queue);

        if ($x == $n - 1 && $y == $n - 1) {
            return $moves;
        }

        $moves++;

        $nextMoves = [
            [$x + $a, $y + $b],
            [$x + $a, $y - $b],
            [$x - $a, $y + $b],
            [$x - $a, $y - $b],
            [$x + $b, $y + $a],
            [$x + $b, $y - $a],
            [$x - $b, $y + $a],
            [$x - $b, $y - $a],
        ];

        foreach ($nextMoves as $next) {
            [$nextX, $nextY] = $next;
            if ($nextX >= 0 && $nextX < $n && $nextY >= 0 && $nextY < $n && !$visited[$nextX][$nextY]) {
                $visited[$nextX][$nextY] = True;
                $queue[] = [$nextX, $nextY, $moves];
            }
        }
    }

    return -1;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));

$result = knightlOnAChessboard($n);

fwrite($fptr, implode("\n", array_map(function($arr) { return implode(' ', $arr); }, $result)) . "\n");

fclose($fptr);
